src/main/scala/com/fulcrumgenomics/umi/ReviewConsensusVariants.scala

```diff
@@ -25,22 +25,99 @@
+    * @param C the count of C observations at the variant locus
+    * @param G the count of G observations at the variant locus
+    * @param T the count of T observations at the variant locus
+    * @param N the count of N observations at the variant locus
+    * @param consensus_read the consensus read name for which the following fields contain values
+    * @param consensus_insert a description of the insert that generated the consensus read
+    * @param consensus_call the base call from the consensus read
+    * @param consensus_qual the quality score from the consensus read
+    * @param a the number of As in raw-reads contributing to the consensus base call at the variant site
+    * @param c the number of Cs in raw-reads contributing to the consensus base call at the variant site
+    * @param g the number of Gs in raw-reads contributing to the consensus base call at the variant site
+    * @param t the number of Ts in raw-reads contributing to the consensus base call at the variant site
+    * @param n the number of Ns in raw-reads contributing to the consensus base call at the variant site
+    */
+  case class ConsensusVariantReviewInfo
+  ( // First set is all the same for each variant
+    chrom: String, pos: Int, ref: String, genotype: String, filters: String, A: Int, C: Int, G: Int, T: Int, N: Int,
+    // Second set is per-consensus reads
+    consensus_read: String, consensus_insert: String, consensus_call: String, consensus_qual: Int, a: Int, c: Int, g: Int, t: Int, n: Int
+  ) extends Metric
+
+  /** Extracts the molecular identifier minus any trailing /A etc. */
+  private[umi] def toMi(r: SAMRecord): String = Option(r.getStringAttribute(ConsensusTags.MolecularId)) match {
+    case Some(mi) =>
+      val slash = mi.lastIndexOf('/')
+      if (slash > 0) mi.substring(0, slash) else mi
+    case None     =>
+      throw new IllegalStateException(s"${r.getReadName} did not have a value for tag ${ConsensusTags.MolecularId}")
+  }
+
+  /** Generate a /1 or /2 suffix based on whether a read is first or second of pair or unpaired. */
+  private[umi] def readNumberSuffix(r : SAMRecord): String = if (r.getReadPairedFlag && r.getSecondOfPairFlag) "/2" else "/1"
+
+  /**
+    * Generates a string of the format 'chr1:500-650 | F1R2' for the insert represented by a read.
+    * Only generates strings for FR pairs, all other reads get "NA".
+    */
+  private[umi] def toInsertString(r: SAMRecord): String = {
+    if (!r.getReadPairedFlag || r.getReadUnmappedFlag || r.getMateUnmappedFlag ||
+      r.getReferenceIndex.toInt != r.getMateReferenceIndex.toInt ||
+      SamPairUtil.getPairOrientation(r) != PairOrientation.FR) {
+      "NA"
+    }
+    else {
+      val outer   = if (r.getReadNegativeStrandFlag) r.getAlignmentEnd else r.getAlignmentStart
+      val isize   = r.getInferredInsertSize
+      val Seq(start, end) = Seq(outer, outer + isize + (if (isize < 0) 1 else -1)).sorted
```

Comment:
This is something else I think we might want to have centralized somewhere - get the start and end of the insert for a paired record with mate on the same chrom, since it's marginally tricky.